src/SimpleSoftwareIO/SMS/SMSServiceProvider.php

```diff
@@ -60,27 +68,103 @@ public function register()
     /**
      * Register the correct driver based on the config file.
      *
-     * @return void
+     * @return CallFireSMS|EmailSMS|EZTextingSMS|MozeoSMS|TwilioSMS
+     * @throws \InvalidArgumentException
      */
     public function registerSender()
     {
-        $this->app['sms.sender'] = $this->app->share(function ($app) {
-            return new DriverManager($app);
-        });
+        $driver = config('sms.driver');
+
+        switch ($driver) {
+            case 'email':
+                return new EmailSMS($this->app['mailer']);
+
+            case 'twilio':
+                return $this->buildTwilio();
+
+            case 'eztexting':
+                return $this->buildEZTexting();
+
+            case 'callfire':
+                return $this->buildCallFire();
+
+            case 'mozeo':
+                return $this->buildMozeo();
+
+            case 'labsmobile':
+                return $this->buildLabsMobile();
+
+            default:
+                throw new \InvalidArgumentException('Invalid SMS driver.');
+        }
     }
 
-    /**
-     * Set a few dependencies on the sms instance.
-     *
-     * @param SMS $sms
-     * @param  $app
-     * @return void
-     */
-    private function setSMSDependencies($sms, $app)
+    protected function buildTwilio()
+    {
+        return new TwilioSMS(
+            new \Services_Twilio(
+                config('sms.twilio.account_sid'),
+                config('sms.twilio.auth_token')
+            ),
+            config('sms.twilio.auth_token'),
+            $this->app['request']->url(),
+            config('sms.twilio.verify')
+        );
+    }
+
+    protected function buildEZTexting()
+    {
+        $provider = new EZTextingSMS(new Client);
+
+        $data = [
+            'User' => config('sms.eztexting.username'),
+            'Password' => config('sms.eztexting.password')
+        ];
+
+        $provider->buildBody($data);
+
+        return $provider;
+    }
+
+    protected function buildCallFire()
     {
-        $sms->setContainer($app);
-        $sms->setLogger($app['log']);
-        $sms->setQueue($app['queue']);
+        $provider = new CallFireSMS(new Client);
+
+        $provider->setUser(config('sms.callfire.app_login'));
+        $provider->setPassword(config('sms.callfire.app_password'));
+
+        return $provider;
+    }
+
+    protected function buildLabsMobile()
```

Comment:
A change was made to the latest master and moved this logic to the DriverManager.  Please update your pull request.